apm-lambda-extension/extension/apm_server.go

```diff
@@ -72,19 +100,75 @@ func PostToApmServer(client *http.Client, agentData AgentData, config *extension
 		req.Header.Add("Authorization", "Bearer "+config.apmServerSecretToken)
 	}
 
+	Log.Debug("Sending data chunk to APM Server")
 	resp, err := client.Do(req)
 	if err != nil {
+		enterBackoff(ctx)
 		return fmt.Errorf("failed to post to APM server: %v", err)
 	}
 
 	//Read the response body
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
+		enterBackoff(ctx)
 		return fmt.Errorf("failed to read the response body after posting to the APM server")
 	}
 
+	apmServerTransportStatus = TransportHealthy
+	Log.Debug("Transport status set to healthy")
 	Log.Debugf("APM server response body: %v", string(body))
 	Log.Debugf("APM server response status code: %v", resp.StatusCode)
 	return nil
 }
+
+func EnqueueAPMData(agentDataChannel chan AgentData, agentData AgentData) {
+	select {
+	case agentDataChannel <- agentData:
+		Log.Debug("Adding agent data to buffer to be sent to apm server")
+	default:
+		Log.Warn("Channel full: dropping a subset of agent request data")
+	}
+}
+
+func IsTransportStatusHealthy() bool {
```

Comment:
Maybe this should be called `IsTransportStatusHealthyOrPending` because I got confused when looking at the tests as this would return true if the status is `pending`

Reply:
Solved in 240fa40